src/cascadia/TerminalCore/TerminalApi.cpp

```diff
@@ -126,6 +126,56 @@ COORD Terminal::GetCursorPosition()
     return newPos;
 }
 
+// Method Description:
+// - deletes uiCount characters starting from the cursor's current position
+// - it moves over the remaining text to 'replace' the deleted text
+// - for example, if the buffer looks like this ('|' is the cursor): [abc|def]
+// - calling DeleteCharacter(1) will change it to: [abc|ef],
+// - i.e. the 'd' gets deleted and the 'ef' gets shifted over 1 space and **retain their previous text attributes**
+// Arguments:
+// - uiCount, the number of characters to delete
+// Return value:
+// - true if succeeded, false otherwise
+bool Terminal::DeleteCharacter(const unsigned int uiCount)
+{
+    SHORT dist;
+    if (!SUCCEEDED(UIntToShort(uiCount, &dist)))
+    {
+        return false;
+    }
+    const auto cursorPos = _buffer->GetCursor().GetPosition();
+    const auto copyToPos = cursorPos;
+    COORD copyFromPos{ cursorPos.X + dist, cursorPos.Y };
+    auto sourceWidth = _mutableViewport.RightExclusive() - copyFromPos.X;
+    SHORT width;
+    if (!SUCCEEDED(UIntToShort(sourceWidth, &width)))
+    {
+        return false;
+    }
+
+    // Get a rectangle of the source
+    auto source = Viewport::FromDimensions(copyFromPos, width, (SHORT)1);

```

Comment:
```suggestion
    auto source = Viewport::FromDimensions(copyFromPos, width, 1s);
```

Reply:
That does not work, does it require some header file to be included that's not being included right now?

Reply:
Looks like the (SHORT) cast was not even necessary. Removed.